File: LAB19/task_4.py
# ...existing code...
def print_students(students):
    if not isinstance(students, list):
        raise TypeError("Expected a list of names")
    for name in students:
        print(name)

if __name__ == "__main__":
    try:
        answer = input("Enter student names (comma-separated). Or press Enter to input one name per line:\n").strip()
        if answer == "":
            print("Enter names one per line. Press Enter on an empty line to finish.")
            names = []
            while True:
                try:
                    line = input().strip()
                except EOFError:
                    break
                if line == "":
                    break
                names.append(line)
            if not names:
                print("No names entered.")
            else:
                print_students(names)
        else:
            students = [s.strip() for s in answer.split(",") if s.strip()]
            if not students:
                print("No names entered.")
            else:
                print_students(students)
    except KeyboardInterrupt:
        print("\nInterrupted.")
# ...existing code...

File: LAB19/task_4.js
// ...existing code...
function printStudents(students) {
    if (!Array.isArray(students)) throw new TypeError("Expected an array of names");
    students.forEach(name => console.log(name));
}

module.exports = { printStudents };

if (require.main === module) {
    const readline = require('readline');
    const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

    rl.question('Enter student names (comma-separated): ', (answer) => {
        const students = answer.split(',').map(s => s.trim()).filter(Boolean);
        if (students.length === 0) {
            console.log('No names entered.');
        } else {
            printStudents(students);
        }
        rl.close();
    });
}
// ...existing code...